pkg/sql/opt_exec_factory.go

```diff
@@ -2026,7 +2023,8 @@ func (ef *execFactory) ConstructAlterRangeRelocate(
 	toStoreID tree.TypedExpr,
 	fromStoreID tree.TypedExpr,
 ) (exec.Node, error) {
-	if !ef.planner.ExecCfg().IsSystemTenant() {
+
+	if relocateSubject != tree.RelocateLease && !ef.planner.ExecCfg().IsSystemTenant() {
```

Comment:
Check relocateSubject != tree.RelocateLease temporarily until https://github.com/cockroachdb/cockroach/issues/93986 removes this block.